### PR TITLE
feat(hooks): add git post-commit hook for backlog task events

### PR DESCRIPTION
**Pull request by jpoley**
## Summary
- Implements automatic event emission when backlog task files are modified through git commits
- Detects task creation, status changes, AC changes, and task completion
- Includes comprehensive integration test suite (9 tests)
- Adds documentation for hook installation and usage

## Changes
- `scripts/hooks/post-commit-backlog-events.sh`: Git post-commit hook that parses task file changes
- `tests/test_post_commit_hook.py`: Integration tests with mock specify CLI
- `docs/guides/backlog-git-hooks.md`: User guide for hook installation

## Code Review Fixes (from PR #793)
- Fixed test_hook_handles_missing_specify to properly test without specify CLI
- Added TASK_ID validation (must match ^task-[0-9.]+$)
- Added comment explaining Git's empty tree hash constant
- Removed reference to non-existent install-hooks.sh script from docs
- Renamed "Trigger" column to "Detection Method" in events table
- Removed obsolete TEST_NOTES.md file

## Test plan
- [x] All 9 integration tests pass (`pytest tests/test_post_commit_hook.py -v`)
- [x] Hook correctly detects new task creation
- [x] Hook correctly detects status changes
- [x] Hook correctly detects task completion (status → Done)
- [x] Hook correctly detects AC checkbox changes
- [x] Hook handles edge cases (missing specify CLI, non-task files, multiple tasks)
- [x] All commits have DCO sign-off

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/test_post_commit_hook.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+    # Should have 3 task.created events
+    created_events = [e for e in events if "task.created" in e]
+
+    assert len(created_events) >= 3, (
+        f"Expected 3 task.created events, got {len(created_events)}: {events}"
+    )
+
+
+def test_hook_parses_task_id_with_dashes(mock_git_repo: Path) -> None:
```
[the remark this answers is not in the document]
Comment (jpoley) [reply tests/test_post_commit_hook.py]: fix me
